observer/plugins/dns/dns.go

```diff
@@ -0,0 +1,43 @@
+package main
+
+import (
+	"fmt"
+	"time"
+
+	odns "github.com/miekg/dns"
+)
+
+// Probe is the exported handler object for monitors configured to use
+// DNS probes
+type Probe struct {
+	QProto   string
+	QRecurse bool
+	QName    string
+	QServer  string
+	QType    uint16
+}
+
+// Do performs the DNS query provided by the monitor settings
+func (p Probe) Do(tick time.Time, timeout time.Duration) (bool, time.Duration) {
```

Comment:
The start time (`tick`) should be retrieved inside this function rather than passed in.

Instead of `timeout`, a `Context` would be somewhat more idiomatic, though I could be convinced on this since we'll just have to convert it back to a timeout for the DNS prober (our DNS package doesn't support contexts super nicely).